molr-mole-core/src/main/java/io/molr/mole/core/logging/MissionLogHandler.java

```diff
@@ -0,0 +1,31 @@
+package io.molr.mole.core.logging;
+
+import io.molr.commons.domain.MissionHandle;
+import io.molr.commons.domain.MissionLog;
+import reactor.core.publisher.Flux;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+// TODO Add javadoc
+public class MissionLogHandler {
+    private static Map<MissionHandle, MissionLogger> missionToLogs = new ConcurrentHashMap<>();
+
+    public static void init(MissionHandle handle) {
+        if (!missionToLogs.containsKey(handle)) {
```

Comment:
Though you use a ConcurrentHashMap, this operation is not atomic! In case case somebody would put something in between your .containsKey() and your put() call, this would give non-deterministic results! As you use a ConcurrentHashMap ... consider using the appropriate methods (e.g. putIfAbsent())

Reply:
... putIfAbsent ... just in case you want to conserve the first put entry... in case you just want to use the latest .. just use put() ;-)